proto/src/determined/api/v1/experiment.proto

```diff
@@ -43,6 +43,23 @@ message GetExperimentResponse {
 
 // Get a list of experiments.
 message GetExperimentsRequest {
+  // filtering by experiment ids
+  message ExperimentFilter {
```

Comment:
I have a question about REST API design here. 
this ends up generating URLs like this: `/api/v1/experiments?sortBy=SORT_BY_ID&orderBy=ORDER_BY_ASC&offset=0&limit=6&archived=false&projectId=104&experimentFilter.restriction=RESTRICTION_EXCLUDE&experimentFilter.experimentIds=125&experimentFilter.experimentIds=126&experimentFilter.experimentIds=34&experimentFilter.experimentIds=105`

it's a bit verbose. some clients / systems sometimes have an upper limit on the URL length (at ~2k characters IIRC). this URL format spends `30+len(experiment_id)` characters per experiment, so it'd hit a limit at 66 experiments which doesn't sound unrealistic to me.

if you were building this REST API without the luggage of grpc-gateway and its enums, how would you want this API URL to look like? 

one possible suggestion here is to do something like this: `/api/v1/experiments?sortBy=SORT_BY_ID&orderBy=ORDER_BY_ASC&offset=0&limit=6&archived=false&projectId=104&experimentIds__in=125,126,34,105`, i.e. "lookup field" and "lookup operation" in a single query parameter, separated by double underscore. 
- for "exclusion" it could look like `experimentIds__notin=125,126,34,105`
- this provides a basis for other operations like `gte` for greater or equal, `lt` for less than, etc. allowing a generic search operations such as `?experimentIds__lte=100&experimentIds__gt=50` for looking up `50 < experiment id <= 100`, as an example.

we currently do not have examples of this approach implemented in the code but it may be worth investing in for the sake of good API.

Reply:
I originally wanted to have the one you suggested `ids=1,2,3`. 
Let me take a look at how REST API is generated

Reply:
In [the documentation ](https://cloud.google.com/endpoints/docs/grpc-service-config/reference/rpc/google.api#rules-for-http-mapping), 
> All other fields are passed via the URL query parameters, and the parameter name is the field path in the request message. A repeated field can be represented as multiple query parameters under the same name.

We might not able to change the style of param query in generated REST API.
I'll search more but if we cant change it all we can do with repeated type is to use shorter variable name or use post as get but its not ideal.

Reply:
push comes to shove, it can be a string in proto that is split by `,` on server side.